### PR TITLE
feat(semantics): Phase 6 - Field & Method Resolution

### PR DESCRIPTION
**Pull request by albertocavalcante**
Implements comprehensive field and method type resolution in the semantic layer,
completing 95% of the semantics migration roadmap and unblocking issue #615 phase 2.

### NativeTypeContext Enhancements
- Add getFieldType() for field resolution with module-level lookup
- Add getMethodReturnType() for method resolution with argument count matching  
- Implement findFieldInModule() and findMethodInModule() helpers
- Enhance fromClassNode() with primitive type detection via isPrimitiveName()

### NativeScope Module Tracking
- Add currentModule: ModuleNode? property for module-level resolution
- Create fromModule() factory method for root scope creation
- Propagate module reference to child scopes

### New Type Calculators (Priority 20)
- FieldNodeCalculator: Resolves FieldNode → declared type
- MethodNodeCalculator: Resolves MethodNode → return type
- ParameterCalculator: Resolves Parameter → declared type

### LSP Integration
- Update GroovySemantics.buildRootScope() to use NativeScope.fromModule()
- Enhance TypeDefinitionProvider with module injection before resolution
- Improve AstPositionQuery to prefer ConstructorCallExpression over type refs

### Unit Tests
- Add NativeTypeContextTest.kt with 31 comprehensive tests
  - 14 happy path scenarios (fields, methods, primitives, generics, arrays)
  - 17 edge cases (missing types, dynamic, null, overloading, special cases)
- All tests passing (31/31)

### Integration Tests  
- Re-enable 4 previously disabled type definition tests
- All TypeDefinition* tests now passing
- Full suite: 1,155/1,156 tests passing (1 intentionally skipped)

## Impact

- Semantic layer: 80% → 95% complete
- Type resolution now supports local module field/method lookups
- Unblocks issue #615 (LSP type definition) phase 2 work
- Three-tier resolution: Native AST → TypeSolver → Workspace Index (deferred)

## References

- Plan: ~/.claude/plans/logical-leaping-goblet.md
- Related: #615 (LSP Type Definition Support)
- Checkpoint: ~/.claude/plans/CHECKPOINT.md

---

> [!NOTE]
> Strengthens semantic type resolution and LSP navigation.
> 
> - **Semantics**: Add module-scoped field/method lookup in `NativeTypeContext` (`getFieldType`, `getMethodReturnType`), with helpers and array `length` handling; improve primitive detection in `fromClassNode`.
> - **Scope/Registry**: Introduce module-aware `NativeScope.fromModule` and propagate to children; register new calculators `FieldNodeCalculator`, `MethodNodeCalculator`, `ParameterCalculator` in `NativeCalculators`.
> - **LSP**: Inject module into semantics in `TypeDefinitionProvider` and add detailed logging; now resolves constructor-call type definitions when class is in the same module.
> - **AST position query**: Heuristics updated in `AstPositionQuery` to prefer `ConstructorCallExpression` over type references; expanded debug tracing.
> - **Tests**: Re-enable type-definition tests, add constructor-call case, and add comprehensive `NativeTypeContextTest` covering field/method resolution and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 486b479389bde054ce1acc9e2968dbbc1e254f68. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * More reliable type-definition navigation (better handling when cursor finds nothing).
  * Improved resolution for constructor calls, fields and methods including primitive/array cases.
  * More accurate distinction between class declarations and type references.

* **Tests**
  * Re-enabled several TypeDefinition tests and added a constructor-call resolution test.
  * Added extensive unit tests covering field and method type resolution across many scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: semantics/native/src/main/kotlin/com/github/albertocavalcante/gvy/semantics/native/NativeTypeContext.kt [hunk 1]
```diff
@@ -38,16 +38,77 @@ class NativeTypeContext(
...
+     * Returns null if not found.
+     */
+    private fun findFieldInModule(classFqn: String, fieldName: String): org.codehaus.groovy.ast.FieldNode? {
+        // Extract class name from FQN
+        val className = classFqn.substringAfterLast('.')
+
+        return scope.currentModule?.classes
+            ?.find { it.name == className }
+            ?.fields
+            ?.find { it.name == fieldName }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply semantics/native/src/main/kotlin/com/github/albertocavalcante/gvy/semantics/native/NativeTypeContext.kt]: Fixed in bbcc3343. Changed to match against full ClassNode.name (FQN) instead of simple class name.

File: parser/native/src/main/kotlin/com/github/albertocavalcante/groovyparser/ast/AstPositionQuery.kt [hunk 2]
```diff
@@ -83,11 +99,40 @@ class AstPositionQuery(private val tracker: NodeRelationshipTracker) {
...
+            }
+            if (logger.isDebugEnabled && filtered.size != candidatesWithoutStatements.size) {
+                logger.debug(
+                    "Filtered out ${candidatesWithoutStatements.size - filtered.size} ClassNode type references",
+                )
+            }
+            filtered.ifEmpty { candidatesWithoutStatements }
+        } else {
+            candidatesWithoutStatements
+        }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply parser/native/src/main/kotlin/com/github/albertocavalcante/groovyparser/ast/AstPositionQuery.kt]: Fixed in bbcc3343. Removed node.lineNumber != 1 guard so isClassDeclaration always validates ClassNode type references.
